### PR TITLE
Feature/wjiang/get short name work around

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
The intel compiler does not deallocate the allocated string in the condition block when optimized compiling flag is used. That leads to crash because the memory is used up. 
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [x] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Comment by weiyuan-jiang** [comment]
Hopefully, this patch [ ](https://github.com/GEOS-ESM/MAPL/tree/feature/wjiang/v2.6.3_patch) can solve your issue temporarily before you upgradeto the latest MAPL.

**Comment by weiyuan-jiang** [comment]
Should this be "exit" ? @tclune 
https://github.com/GEOS-ESM/MAPL/blob/eb3eebb18c47299e3aa87879ac49feeb7a835229/generic/MAPL_Generic.F90#L7087

**Comment by weiyuan-jiang** [comment]
Should we add if (i == j) cycle between the lines? @tclune 
https://github.com/GEOS-ESM/MAPL/blob/eb3eebb18c47299e3aa87879ac49feeb7a835229/generic/MAPL_Generic.F90#L7098-L7099

File: generic/VarConn.F90 [hunk 1]
```diff
@@ -147,6 +147,7 @@ logical function checkUnused(this)
       integer                               :: I
       class(Logger), pointer :: lgr
       type(VarConnType), pointer :: conn
+      character(len=ESMF_MAXSTR)            :: from_name, to_name
```
Comment (tclune) [inline generic/VarConn.F90:150]: Of course part of the functionality of pFlogger is to _avoid_ needing to declare additional variables like this.  Sigh.

**Review by tclune — DISMISSED** [review]

**Comment by tclune** [comment]
> Should this be "exit" ? @tclune
> 
> https://github.com/GEOS-ESM/MAPL/blob/eb3eebb18c47299e3aa87879ac49feeb7a835229/generic/MAPL_Generic.F90#L7087

Really the question should go to @atrayano    I'm wary of changing the logic in such a fundamental way.  Why do you think it should be `exit`?

**Comment by tclune** [comment]
> Should we add if (i == j) cycle between the lines? @tclune
> 
> https://github.com/GEOS-ESM/MAPL/blob/eb3eebb18c47299e3aa87879ac49feeb7a835229/generic/MAPL_Generic.F90#L7098-L7099

Again really a question for @atrayano

**Review by mathomp4 — APPROVED** [review]
Passes with GCM.

**Comment by mathomp4** [comment]
This passed the GCM. Once the CI is happy, I'll merge and make a dev to main pr for release.

**Pull request merged**